data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -293,6 +295,13 @@ private CloseableIterable<Record> openDeletes(DeleteFile deleteFile, Schema dele
           builder.filter(Expressions.equal(MetadataColumns.DELETE_FILE_PATH.name(), filePath));
         }
 
+        if (deleteFile.keyMetadata() != null) {
+          EncryptionKeyMetadata keyMetadata =
+              EncryptionUtil.parseKeyMetadata(deleteFile.keyMetadata());
```

Comment:
Cast or parse would be better here, too.

Reply:
keyMetadata() is always a ByteBuffer here, so it has to be parsed.

Reply:
Thanks for clarifying. That makes sense.